### ISSUE TITLE
Bug: Flagging the 'key' property as an unknown property

### ISSUE THREAD

**Issue by leechongren** [eslint/eslint#16273]
### Environment

Node version: 14.4.0
npm version: 6.14.5
Local ESLint version: 8.23.0
Global ESLint version: 8.23.0
Operating System: Windows

### What parser are you using?

Other

### What did you do?

![bug-report](https://user-images.githubusercontent.com/59591074/188255270-69478480-4f58-4d2a-86d6-7484f413a352.JPG)

```js

```

### What did you expect to happen?

It should not flag it as error because key is not an unknown property.

### What actually happened?

It flagged the key as an unknown property

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by JustFly1984** [comment]
In continuation of this issue, can we highlight the key properties, which are not in the map loop? I'm frequently noticing key properties moved outside of the loop in refactoring process, and forgotten.

**Comment by leechongren** [comment]
Apologies, I realised the actual issue is actually in another library, `eslint-plugin-react`, and it was reported there already so i am closing this. They are currently working on a fix so it should be resolved soon.

https://github.com/jsx-eslint/eslint-plugin-react/issues/3380
